src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -9629,6 +9629,102 @@ static void Main()
             comp.VerifyDiagnostics(expectedDiagnostics);
         }
 
+        [Fact]
+        [WorkItem(58106, "https://github.com/dotnet/roslyn/issues/58106")]
+        public void InferDelegateType_01()
```

Comment:
Are these enough of a stress test that it will demonstrate the perf problem? Or can we create a pathological test that won't compile in any reasonable amount of time without this change?

Reply:
No, these are not stress tests. I decided to test these cases more directly by tracking the `InferredDelegateCount` instead. We could test perf although that might require a test that is slow even after the fix.

Reply:
I like tracking the count here as that is directly tracking the problem we are trying to fix here (avoiding unnecessary work). 

My biggest fear is making sure that future developers understand why we are tracking the count though. The only documentation here is the issue and the issue doesn't specifically call out that this is a customer scenario. It only mentions that it's causing us to do unnecessary work. I think we should make the issue more specific here, say it caused customer reported problems and link to the VS feedback (or link to the feedback here). 

Want to make it clear as possible to future developers why this count is meaningful.

Reply:
I've added comments to these tests and links to the related issue.